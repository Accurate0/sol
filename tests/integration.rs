use assert_cmd::cargo::CommandCargoExt;
use insta::assert_snapshot;
use rstest::rstest;
use std::{path::PathBuf, process::Command};

#[rstest]
fn run_success(#[files("tests/files/success/*.rl")] path: PathBuf) {
    let mut cmd = Command::cargo_bin(env!("CARGO_PKG_NAME")).unwrap();
    let cmd = cmd
        .arg("run")
        .arg(&path)
        .env("NO_COLOR", "true")
        .env("PLRS_LOG", "info");

    let output = cmd.output().unwrap();

    let snapshot_name = format!("success__{}", path.file_name().unwrap().to_string_lossy());

    let output = format!(
        "{}\n\n{}",
        String::from_utf8(output.stdout).unwrap(),
        String::from_utf8(output.stderr).unwrap()
    );

    assert_snapshot!(snapshot_name, output);
}

#[rstest]
fn run_fail(#[files("tests/files/fail/*.rl")] path: PathBuf) {
    let mut cmd = Command::cargo_bin(env!("CARGO_PKG_NAME")).unwrap();
    let cmd = cmd
        .arg("run")
        .arg(&path)
        .env("NO_COLOR", "true")
        .env("PLRS_LOG", "info");

    let output = cmd.output().unwrap();

    let snapshot_name = format!("fail__{}", path.file_name().unwrap().to_string_lossy());

    let output = format!(
        "{}\n\n{}",
        String::from_utf8(output.stdout).unwrap(),
        String::from_utf8(output.stderr).unwrap()
    );

    assert_snapshot!(snapshot_name, output);
}
