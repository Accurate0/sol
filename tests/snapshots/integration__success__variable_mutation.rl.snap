---
source: tests/integration.rs
expression: output
---
 INFO plrs: CompiledProgram {
    functions: [],
    global_code: [
        LoadLiteral {
            dest: 1,
            src: 0,
        },
        LoadLiteral {
            dest: 2,
            src: 1,
        },
        LoadLiteral {
            dest: 3,
            src: 2,
        },
        Add {
            dest: 4,
            lhs: 2,
            rhs: 3,
        },
    ],
    global_register_count: 5,
    literals: [
        Integer(
            1,
        ),
        Integer(
            2,
        ),
        Integer(
            3,
        ),
    ],
}
 INFO plrs::vm: executing: LoadLiteral { dest: 1, src: 0 }
 INFO plrs::vm: 1 Integer(1)

 INFO plrs::vm: executing: LoadLiteral { dest: 2, src: 1 }
 INFO plrs::vm: 1 Integer(1)
 INFO plrs::vm: 2 Integer(2)

 INFO plrs::vm: executing: LoadLiteral { dest: 3, src: 2 }
 INFO plrs::vm: 1 Integer(1)
 INFO plrs::vm: 2 Integer(2)
 INFO plrs::vm: 3 Integer(3)

 INFO plrs::vm: executing: Add { dest: 4, lhs: 2, rhs: 3 }
 INFO plrs::vm: 1 Integer(1)
 INFO plrs::vm: 2 Integer(2)
 INFO plrs::vm: 3 Integer(3)
 INFO plrs::vm: 4 Integer(5)
